src/flow/src/transform/literal.rs

```diff
@@ -124,25 +132,21 @@ pub(crate) fn from_substrait_literal(lit: &Literal) -> Result<(Value, CDT), Erro
         },
         Some(LiteralType::Fp32(f)) => (Value::from(*f), CDT::float32_datatype()),
         Some(LiteralType::Fp64(f)) => (Value::from(*f), CDT::float64_datatype()),
-        Some(LiteralType::Timestamp(t)) => match lit.type_variation_reference {
-            TIMESTAMP_SECOND_TYPE_VARIATION_REF => (
-                Value::from(Timestamp::new_second(*t)),
-                CDT::timestamp_second_datatype(),
-            ),
-            TIMESTAMP_MILLI_TYPE_VARIATION_REF => (
-                Value::from(Timestamp::new_millisecond(*t)),
-                CDT::timestamp_millisecond_datatype(),
-            ),
-            TIMESTAMP_MICRO_TYPE_VARIATION_REF => (
-                Value::from(Timestamp::new_microsecond(*t)),
-                CDT::timestamp_microsecond_datatype(),
-            ),
-            TIMESTAMP_NANO_TYPE_VARIATION_REF => (
-                Value::from(Timestamp::new_nanosecond(*t)),
-                CDT::timestamp_nanosecond_datatype(),
-            ),
-            others => not_impl_err!("Unknown type variation reference {others}",)?,
-        },
+        Some(LiteralType::Timestamp(t)) => (
+            Value::from(Timestamp::new_microsecond(*t)),
+            CDT::timestamp_microsecond_datatype(),
+        ),
+        Some(LiteralType::PrecisionTimestamp(prec_ts)) => {
+            let (prec, val) = (prec_ts.precision, prec_ts.value);
```

Comment:
Nit(Nonblocking cc @discord9 ): Could we define an enum for timestamp precision, like:

```
#[derive(Debug)]
enum TimestampPrecision {
    Second = 0,
    Millisecond = 3,
    Microsecond = 6,
    Nanosecond = 9,
}

impl TryFrom<i32> for TimestampPrecision {
    type Error = Error;

    fn try_from(prec: i32) -> Result<Self, Self::Error> {
        match prec {
            0 => Ok(Self::Second),
            3 => Ok(Self::Millisecond),
            6 => Ok(Self::Microsecond),
            9 => Ok(Self::Nanosecond),
            _ => not_impl_err!("Unsupported precision: {prec}"),
        }
    }
}
```